File: php/nearestDriverDistance.php
<?php

function nearestDriverDistance(){
    $url = "https://logistic.groupngs.com/api/";
    $private_key = '';

    $headers = array(
        'Authorization: Bearer '.$private_key,
        'Content-Type: application/json'
    );

    $body = array(
        "action" => "nearestDriverDistance",
        "countryCode" => "UG",
        "vehicleType" => "DELIVERY_CAB",
        "position" => [0.315133, 32.576353]
    );

    $req = curl_init($url);
    curl_setopt_array($req, array(
        CURLOPT_POST => TRUE,
        CURLOPT_RETURNTRANSFER => TRUE,
        CURLOPT_HTTPHEADER => $headers,
        CURLOPT_POSTFIELDS => json_encode($body)
    ));

    // Send request 
    $res = curl_exec($req);

    // checking errors
    if($res === FALSE){
        die(curl_error($req));
    }

    // get response
    echo json_decode($req, TRUE);
    
    //close
    curl_close($req);
}


?>

File: python/nearest_driver_distance.py
import requests

def nearest_driver_distance():
    private_key = ''
    url = 'https://logitist.groupngs.com/api/'

    body = {
        "action": "nearestDriverDistance",
        "countryCode": "UG",
        "vehicleType": "DELIVERY_CAB",
        "position": "[0.315133, 32.576353]"
    }

    headers = {
        'Authorization': 'Bearer ' + private_key
    }

    req = requests.post(url, data=body, headers=headers)
    req.raise_for_status()
    print(req.json())